tests/integration_tests/performance/test_snapshot_perf.py

```diff
@@ -247,180 +238,68 @@ def snapshot_resume_producer(logger, vm_builder, snapshot, snapshot_type, use_ra
-                snapshot_type,
-                target_version,
-                context.microvm.name(),
-                context.kernel.name(),
-                context.disk.name(),
-            )
-        )
-
-        # Create a fresh microVM from artifacts.
-        vm_instance = vm_builder.build(
-            kernel=context.kernel,
-            disks=[rw_disk],
-            ssh_key=ssh_key,
-            config=context.microvm,
-            diff_snapshots=diff_snapshots,
-            use_ramdisk=True,
-        )
-        vm = vm_instance.vm
-        # Configure metrics system.
-        metrics_fifo_path = os.path.join(vm.path, "metrics_fifo")
-        metrics_fifo = log_tools.Fifo(metrics_fifo_path)
-
-        response = vm.metrics.put(
-            metrics_path=vm.create_jailed_resource(metrics_fifo.path)
-        )
-        assert vm.api_session.is_status_no_content(response.status_code)
-
-        vm.start()
-
-        # Check if the needed CPU cores are available. We have the API
-        # thread, VMM thread and then one thread for each configured vCPU.
-        assert CpuMap.len() >= 2 + vm.vcpus_count
-
-        # Pin uVM threads to physical cores.
-        current_cpu_id = 0
-        assert vm.pin_vmm(current_cpu_id), "Failed to pin firecracker thread."
-        current_cpu_id += 1
-        assert vm.pin_api(current_cpu_id), "Failed to pin fc_api thread."
-        for idx_vcpu in range(vm.vcpus_count):
-            current_cpu_id += 1
-            assert vm.pin_vcpu(
-                idx_vcpu, current_cpu_id + idx_vcpu
-            ), f"Failed to pin fc_vcpu {idx_vcpu} thread."
-
-        st_core = core.Core(
-            name="snapshot_create_full_latency"
-            if snapshot_type == SnapshotType.FULL
-            else "snapshot_create_diff_latency",
-            iterations=SAMPLE_COUNT,
-        )
-
-        prod = producer.LambdaProducer(
-            func=snapshot_create_producer,
-            func_kwargs={
-                "logger": logger,
-                "vm": vm,
-                "disks": [rw_disk],
-                "ssh_key": ssh_key,
-                "target_version": target_version,
-                "metrics_fifo": metrics_fifo,
-                "snapshot_type": snapshot_type,
-            },
-        )
-
-        cons = consumer.LambdaConsumer(
-            func=lambda cons, result: cons.consume_stat(
-                st_name="max", ms_name="latency", value=result
-            ),
-            func_kwargs={},
-        )
-        eager_map(
-            cons.set_measurement_def,
-            snapshot_create_measurements(context.microvm.name(), snapshot_type),
-        )
-
-        st_core.add_pipe(producer=prod, consumer=cons, tag=context.microvm.name())
-
-    # Gather results and verify pass criteria.
-    try:
-        result = st_core.run_exercise()
-    except core.CoreException as err:
-        handle_failure(file_dumper, err)
-
-    file_dumper.dump(result)
-
+@pytest.mark.parametrize("io_engine", ENGINES)
```

Comment:
Same for this. Does it make sense to make `io_engine` a fixture?

Reply:
done in latest revision